From 3ff801b4014d35941e4145f24b5bdcd844926b4a Mon Sep 17 00:00:00 2001
From: Hu Wang <huw@codeaurora.org>
Date: Thu, 8 Nov 2018 19:58:10 +0800
Subject: [PATCH 2/2] wifi: Add wifi_set_latency_level() to legacy wifi hal

This is used to set wifi latency level based on use scenario at
runtime.

CRs-Fixed: 2371216
Change-Id: Ibbbbc81fc1813e02194c6c97975de91570b56192
---
 include/hardware_legacy/wifi_config.h | 3 +++
 include/hardware_legacy/wifi_hal.h    | 1 +
 2 files changed, 4 insertions(+)

diff --git a/include/hardware_legacy/wifi_config.h b/include/hardware_legacy/wifi_config.h
index d3e8f97..b3062e0 100644
--- a/include/hardware_legacy/wifi_config.h
+++ b/include/hardware_legacy/wifi_config.h
@@ -25,6 +25,9 @@ wifi_error wifi_set_country_code(wifi_interface_handle iface, const char* countr
 //Set the powersave to driver
 wifi_error wifi_set_qpower(wifi_interface_handle iface, u8 powersave);
 
+//Set the latency level to driver
+wifi_error wifi_set_latency_level(wifi_interface_handle iface, u16 level);
+
 //set the wifi_iface stats averaging factor used to calculate
 // statistics like average the TSF offset or average number of frame leaked
 // For instance, upon beacon reception:
diff --git a/include/hardware_legacy/wifi_hal.h b/include/hardware_legacy/wifi_hal.h
index 22ea71e..0c7e6ae 100644
--- a/include/hardware_legacy/wifi_hal.h
+++ b/include/hardware_legacy/wifi_hal.h
@@ -468,6 +468,7 @@ typedef struct {
     wifi_error (*wifi_add_or_remove_virtual_intf)(wifi_interface_handle iface,
                                                   const char* ifname, u32 iface_type,
                                                   bool create);
+    wifi_error (*wifi_set_latency_level)(wifi_interface_handle iface, u16 level);
 } wifi_hal_fn;
 wifi_error init_wifi_vendor_hal_func_table(wifi_hal_fn *fn);
 #ifdef __cplusplus
-- 
2.17.1

